valhalla/sif/dynamiccost.h

```diff
@@ -217,6 +217,19 @@ class DynamicCost {
     return (node->access() & access_mask_) || ignore_access_;
   }
 
+  /**
+   * Checks if access information was initially specified in the data.
+   * If the node didn't have any access specifications then it can be restricted.
+   * @param   node  Pointer to node information.
+   * @return  Returns true if access was specified for the node and access is allowed.
+   */
+  virtual bool InitiallyAllowed(const baldr::NodeInfo* node) const {
```

Comment:
what is the purpose of this method ? how is supposed to be used ?

Reply:
i'll add that we must be very careful when considering expanding the scope of the costing interface and we should consider existing semantics and algorithms that make use of those and if perhaps they already could handle the functionality we are looking to achieve

from the title and a bit of looking at the diff i see that we are basically trying keep track of whether or not access was "tagged" or "inferred" for barrier nodes. i guess the idea is that if its tagged as access yes then we dont need to penalize it. if that is the case it seems to me all we need to do is:

1. make the changes to nodeinfo to have the extra bit that says the access is tagged (i would consider naming it as such as well since we have the same concept for speeds on ways being "tagged")
2. modify base_transition_cost to check that flag on the nodeinfo and not penalize if its there

Reply:
The main idea was to penalise the gates with no access information and additionally the gates that have restricted access. But since the fine does not affect the restricted nodes I agree that it is clearer to use the flag directly.